internal/app/api/balance/handler.go

```diff
@@ -0,0 +1,168 @@
+package balance
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"github.com/go-chi/chi/v5"
+	"github.com/korol8484/gofermart/internal/app/api/util"
+	"github.com/korol8484/gofermart/internal/app/domain"
+	"go.uber.org/zap"
+	"io"
+	"net/http"
+	"time"
+)
+
+type listResponse struct {
+	Order       string  `json:"order"`
+	Sum         float64 `json:"sum"`
+	ProcessedAt string  `json:"processed_at"`
+}
+
+type balanceResponse struct {
+	Current float64 `json:"current"`
+	Sum     float64 `json:"withdrawn"`
+}
+
+type withdrawRequest struct {
+	Order string  `json:"order"`
+	Sum   float64 `json:"sum"`
+}
+
+type usecase interface {
+	UserWithdrawals(ctx context.Context, userID domain.UserID) ([]*domain.Balance, error)
+	GetUserSumWC(ctx context.Context, userID domain.UserID) (*domain.SumWC, error)
+	Withdraw(ctx context.Context, userID domain.UserID, number string, sum float64) (*domain.Balance, error)
+}
+
+type Handler struct {
+	uc  usecase
+	log *zap.Logger
+}
+
+func NewBalanceHandler(uc usecase, log *zap.Logger) *Handler {
+	return &Handler{uc: uc, log: log}
+}
+
+func (h *Handler) list(w http.ResponseWriter, r *http.Request) {
+	userID, ok := util.UserIDFromContext(r.Context())
+	if !ok {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	withdrawals, err := h.uc.UserWithdrawals(r.Context(), userID)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if len(withdrawals) < 1 {
```

Comment:
Думаю такое сравнение будет нагляднее:

```suggestion
	if len(withdrawals) == 0 {
```